minifi-toolkit/minifi-toolkit-configuration/src/main/java/org/apache/nifi/minifi/toolkit/configuration/ConfigMain.java

```diff
@@ -235,62 +252,127 @@ private static void addRemoteProcessGroupPortDTOs(Map<String, String> connectabl
         }
     }
 
+    public int upgrade(String[] args) {
+        if (args.length != 3) {
+            printUgradeUsage();
+            return ERR_INVALID_ARGS;
+        }
+
+        ConfigSchema configSchema = null;
+        try (InputStream inputStream = pathInputStreamFactory.create(args[1])) {
+            try {
+                configSchema = SchemaLoader.loadConfigSchemaFromYaml(inputStream);
```

Comment:
@JPercivall for the latest schema, the convert will just return this.  I can add extra logic to detect that case and do nothing but it seems like that's harmless behavior and could be used to round-trip and/or reformat the file.